modules/3d/test/test_tsdf.cpp

```diff
@@ -1048,6 +1204,13 @@ TEST(HashTSDF_CPU, valid_points_common_framesize_frame)
     cv::ocl::setUseOpenCL(true);
 }
 
+TEST(HashTSDF_CPU, reproduce_volPoseRot)
+{
+    cv::ocl::setUseOpenCL(false);
+    regressionVolPoseRot();
+    cv::ocl::setUseOpenCL(true);
```

Comment:
The `regressionVolPoseRot` in this unit uses `cv::Mat`, so it's always CPU only, right?

Reply:
No, a `Volume` class behaves differently.
If OpenCL is available and enabled then the `Volume` is created with GPU version used. Otherwise the CPU version is used. Once created, the type of `Volume` object cannot be changed.
Then a data is obtained from `InputArray` or `OutputArray` using `getMat()` or `getUMat()` methods regarding what `Volume` version is used.